synapse/metrics/background_process_metrics.py

```diff
@@ -101,9 +101,13 @@ def collect(self):
             labels=["name"],
         )
 
-        # We copy the dict so that it doesn't change from underneath us
+        # We copy the dict so that it doesn't change from underneath us.
+        # We also copy the process lists as that can also change
         with _bg_metrics_lock:
-            _background_processes_copy = dict(_background_processes)
+            _background_processes_copy = {
+                k: list(v)
+                for k, v in six.iteritems(_background_processes)
```

Comment:
what is to say that this will be thread-safe when the loop below was not?

Reply:
We're copying the dict and lists from inside a lock

Reply:
ffs. I should probably learn to read.

Reply:
Glad I'm not the only one :grinning: